python/lsst/source/injection/inject_base.py

```diff
@@ -0,0 +1,636 @@
+        doc="Size of the pixel padding surrounding the image. Only those synthetic sources with a centroid "
+        "falling within the ``image + trim_padding`` region will be considered for source injection.",
+        default=100,
+        optional=True,
+    )
+    subset_by_select = Field[bool](
+        doc="If True, limit the input catalog to a subset of sources flagged as True in the *col_select* "
+        "column.",
+        default=False,
+    )
+
+    # General configuration options.
+    mask_plane_name = Field[str](
+        doc="Name assigned to the injected mask plane which is attached to the output exposure.",
+        default="INJECTED",
+    )
+    calib_flux_radius = Field[float](
+        doc="Aperture radius (in pixels) that was used to define the calibration for this image+catalog. "
+        "This will be used to produce the correct instrumental fluxes within the radius. "
+        "This value should match that of the field defined in ``slot_CalibFlux_instFlux``.",
+        default=12.0,
+    )
+    fits_alignment = ChoiceField[str](
+        doc="How should injections from FITS files be aligned?",
+        dtype=str,
+        allowed={
+            "wcs": (
+                "Input image will be transformed such that the local WCS in the FITS header matches the "
+                "local WCS in the target image. I.e., North, East, and angular distances in the input image "
+                "will match North, East, and angular distances in the target image."
+            ),
+            "pixel": (
+                "Input image will **not** be transformed. Up, right, and pixel distances in the input image "
+                "will match up, right and pixel distances in the target image."
+            ),
+        },
+        default="pixel",
+    )
+    stamp_prefix = Field[str](
+        doc="String to prefix to the entries in the *col_stamp* column, for example, a directory path.",
+        default="",
+    )
+    draw_size_scale = Field[float](
+        doc="Multiplicative scale factor applied to the size of the injected box drawn for each source.",
+        default=1.0,
+    )
+    draw_size_max = Field[int](
+        doc="Maximum allowed pixel size of the injected box drawn for each source. "
+        "Clipped to this value if set greater than zero.",
+        default=10000,
+    )
+
+    # Custom column names.
+    col_ra = Field[str](
+        doc="Column name for right ascension (in degrees).",
+        default="ra",
+    )
+    col_dec = Field[str](
+        doc="Column name for declination (in degrees).",
+        default="dec",
+    )
+    col_mag = Field[str](
+        doc="Column name for magnitude.",
+        default="mag",
+    )
+    col_source_type = Field[str](
+        doc="Column name for the source type used in the input catalog. Must match one of the surface "
+        "brightness profiles defined by GalSim.",
+        default="source_type",
+    )
+    col_select = Field[str](
```

Comment:
Having spoken with member of AP who will be using this code base, I don't think it's trivial for them to define boolean selections based on catalog logic alone. They have template types that are defined elsewhere, and then that logic gets encoded into this boolean column.

I suppose we could replace it with your evaluation logic, and then AP could still include a column named `select` which could be selected against?

Let me look at implementing this, and I'll report back on this thread later today.

Reply:
I think we're in agreement now that this can be implemented here and should maintain current behavior for legacy fakes users. The only issue is that end-users may need to know some of our internal names for certain objects. The flexibility granted via this change however is likely worth it, as long as this feature is well documented. 

The changes suggested above will be implemented on this ticket, and documentation ticket [DM-39728](https://jira.lsstcorp.org/browse/DM-39728) will document them.

Reply:
A new `selection` config option has now been added. This replaces the old `subset_by_select` on/off config option, and the `col_select` column name config option.

Reply:
PS - this `selection` logic now also covers the prior visit selection logic as well.